src/main/java/org/zalando/fahrschein/NakadiReader.java

```diff
@@ -265,10 +264,22 @@ public void run(long timeout, TimeUnit timeoutUnit) throws IOException {
         }
     }
 
+    private void logException(final IOException e, final int errorCount) {
+        if (errorCount == 0 && e instanceof StreamClosedException) {
+            if (LOG.isTraceEnabled()) {
+                LOG.info("Stream was closed", e);
+            } else {
+                LOG.info("Stream was closed");
+            }
+        } else {
+            LOG.warn("Got [{}] while reading events", e.getClass().getSimpleName(), e);
+        }
+    }
+
     private void expectField(JsonParser jsonParser, String expectedFieldName) throws IOException {
         final String fieldName = jsonParser.getCurrentName();
         if (fieldName == null) {
-            throw new IOException("Stream was closed or no field at current position");
+            throw new IOException("No field at current position");
```

Comment:
I'm no longer sure if this can happen at all, if Jackson reaches eof in the middle of the input it should throw

    JsonParseException: Unexpected end-of-input: ...